pkg/endpoint/policy.go

```diff
@@ -109,7 +109,14 @@ func (e *Endpoint) removeOldFilter(owner Owner, labelsMap *identityPkg.IdentityC
 	port := uint16(filter.Port)
 	proto := uint8(filter.U8Proto)
 
-	for _, sel := range filter.FromEndpoints {
+	fromEndpointsSelectors := filter.FromEndpoints
```

Comment:
Why not use `filter.FromEndpoints` directly? ohh I see, you're adding a L3 wildcard if L3 is not defined. Can you please add a comment for that?